packages/common/src/components/Preview/index.tsx

```diff
@@ -182,15 +182,31 @@ class BasePreview extends React.Component<Props, State> {
     }
   };
 
-  currentUrl = () =>
-    this.props.url ||
-    (this.serverPreview
-      ? getSSEUrl(this.props.sandbox, this.props.initialPath)
-      : frameUrl(
-          this.props.sandbox,
-          this.props.initialPath || '',
-          this.state && this.state.useFallbackDomain
-        ));
+  currentUrl = () => {
```

Comment:
I think this function needs to be a bit different, there are two issues I see:

* It now returns `this.props.url` by default, even if it isn't set (last statement)
* It tries to append the url to sseURL or frameURL, but sometimes these urls already have something appended (thanks to `initialPath`).
* There is some double logic (the check on this.serverPreview is twice)

I think it can be simplified to:

```ts
currentUrl = () => {
  const { url, sandbox } = this.props;
  if (url && !url.startsWith('/')) {
    // An absolute url is given, just return that
    return url;
  }

  // url may be a relative path (/test), so start with that
  const initialPath = url || this.props.initialPath || '';

  return this.serverPreview
      ? getSSEUrl(sandbox, initialPath)
      : frameUrl(sandbox, initialPath, this.state && this.state.useFallbackDomain);
}
```

Reply:
Seems to work great and changed